R/utils.R

```diff
@@ -0,0 +1,23 @@
+
+check_color <- function(clr) {
```

Comment:
I think in this function let's just check if `clr` is NULL, and if so, just return NULL, and don't do the rest of the checks here. That will then I think allow a user to keep using say without having multicolor installed.

Reply:
now that we only `multicolor::insert_rainbow` after the `check4pkg` call that you suggested ^, if `clr` is NULL then `check_color` returns NULL since both of the top-level conditionals evaluate to false. 

```
check_color <- function(clr) {
  if (!is.null(clr) && !all(inherits(clr, c("crayon", "character")))) {
    stop("All colors must be of class character or crayon",
         call. = FALSE)
  }
  
  if (all(inherits(clr, "character"))) {
    if (length(clr) > 1 || clr == "rainbow") {
      check4pkg("multicolor")
      
      # Change "rainbow" into roygbiv
      clr <- multicolor::insert_rainbow(clr)
      
      if (!all(is.character(clr))) {
        stop("If color arguments have > length 1, all colors must be of class character",
             call. = FALSE)
      }
    }
  }
  return(clr)
}
```

so they should only need multicolor installed if they supply multiple colors or a "rainbow"

```
library(cowsay)

say()
#> 
#>  -------------- 
#> Hello world! 
#>  --------------
#>     \
#>       \
#>         \
#>             |\___/|
#>           ==) ^Y^ (==
#>             \  ^  /
#>              )=*=(
#>             /     \
#>             |     |
#>            /| | | |\
#>            \| | |_|/\
#>       jgs  //_// ___/
#>                \_)
#> 
```

I'm fine explicitly checking if `clr` is NULL and returning NULL earlier to be more explicit about it, though

Reply:
ah okay, makes sense then